geps/gep-1762.md

```diff
@@ -0,0 +1,294 @@
+	Labels map[string]string `json:"labels,omitempty"`
+	// Annotations that should be applied to any resources created in response to this Gateway.
+	//
+	// For implementations creating other Kubernetes objects, this should be the `metadata.annotations` field on resources.
+	// For other implementations, this refers to any relevant (implementation specific) "annotations" concepts.
+	//
+  // If Annotations is set on the GatewayClass as well, the labels are merged with the Gateway's labels taking precedence.
+  //
+	// Support: Implementation-specific
+	Annotations map[string]string `json:"annotations,omitempty"`
+	// ParametersRef is a reference to a resource that contains the configuration
+	// parameters corresponding to the Gateway. This is optional if the
+	// controller does not require any additional configuration.
+	//
+	// ParametersRef can reference a standard Kubernetes resource, i.e. ConfigMap,
+	// or an implementation-specific custom resource. The resource must be namespace-scoped
+  // and live in the same namespace.
+	//
+  // If ParametersRef is set on both a Gateway and the corresponding GatewayClass, the merging
+  // behavior between the two are implementation-specific.
+  //
+	// Support: Implementation-specific
+	//
+	// +optional
+	ParametersRef *LocalParametersReference `json:"parametersRef,omitempty"`
+	...
+}
+
+type GatewayClassInfrastructure struct {
+	// Labels that should be applied to any resources created in response to this GatewayClass.
+	//
+	// For implementations creating other Kubernetes objects, this should be the `metadata.labels` field on resources.
+	// For other implementations, this refers to any relevant (implementation specific) "labels" concepts.
+  //
+  // If Labels is set on the Gateway as well, the labels are merged with the Gateway's labels taking precedence.
+	//
+	// Support: Implementation-specific
+	Labels map[string]string `json:"labels,omitempty"`
+	// Annotations that should be applied to any resources created in response to this GatewayClass.
+	//
+	// For implementations creating other Kubernetes objects, this should be the `metadata.annotations` field on resources.
+	// For other implementations, this refers to any relevant (implementation specific) "annotations" concepts.
+	//
+  // If Annotations is set on the Gateway as well, the labels are merged with the Gateway's annotations taking precedence.
+  //
+	// Support: Implementation-specific
+	Annotations map[string]string `json:"annotations,omitempty"`
+	...
+}
+
+// LocalParametersReference identifies an API object containing controller-specific
+// configuration resource within the cluster.
+type LocalParametersReference struct {
+	// Group is the group of the referent.
+	Group Group `json:"group"`
+
+	// Kind is kind of the referent.
+	Kind Kind `json:"kind"`
+
+	// Name is the name of the referent.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name"`
+}
```

Comment:
I don't think this is needed anymore.

Reply:
Its in GatewayInfrastructure, so probably pending decision on https://github.com/kubernetes-sigs/gateway-api/pull/1757#discussion_r1327657519?

Reply:
Yeah, I'd personally rather keep this GEP as focused as possible since we're really only a couple of weeks away from trying to be feature complete for v1.0. I don't really have anything against ParamsRef here, but similar to the comment above, it introduces some form of merging logic between GWC and GW, and I want to make sure we take the time to get it right so would rather leave it out of scope for now.